### PR TITLE
add AsSpan and AsMemory benchmarks

### PR DESCRIPTION
**Pull request by adamsitnik**
Benchmarks for https://github.com/dotnet/coreclr/issues/21594

### CODE REVIEW

File: src/benchmarks/micro/corefx/System.Memory/Constructors.cs [hunk 1]
```diff
@@ -76,6 +76,12 @@ public Constructors()
         [Benchmark]
         public System.ReadOnlyMemory<T> ReadOnlyMemoryFromArrayStartLength() => new System.ReadOnlyMemory<T>(_nonEmptyArray, start: 0, length: Size);
 
+        [Benchmark]
+        public System.Span<T> ArrayAsSpan() => _nonEmptyArray.AsSpan();
+        
+        [Benchmark]
+        public System.Memory<T> ArrayAsMemory() => _nonEmptyArray.AsMemory();
```
Comment (stephentoub) [inline src/benchmarks/micro/corefx/System.Memory/Constructors.cs]: What about with arguments?

File: src/benchmarks/micro/corefx/System.Memory/Constructors.cs [hunk 2]
```diff
@@ -76,6 +76,12 @@ public Constructors()
         [Benchmark]
         public System.ReadOnlyMemory<T> ReadOnlyMemoryFromArrayStartLength() => new System.ReadOnlyMemory<T>(_nonEmptyArray, start: 0, length: Size);
 
+        [Benchmark]
+        public System.Span<T> ArrayAsSpan() => _nonEmptyArray.AsSpan();
```
Comment (jkotas) [inline src/benchmarks/micro/corefx/System.Memory/Constructors.cs]: This does not cover the case that regressed in linked issue. The case that regressed was `AsSpan` overload that takes the index and length arguments.

Comment (danmosemsft) [reply src/benchmarks/micro/corefx/System.Memory/Constructors.cs]: Also this is an array of 10 elements. @Anipik 's case was 100_000 elements. They may have different perf characteristics.

**Comment by adamsitnik** [comment]
@stephentoub I did not have the time to rebuild and re-run the benchmarks with and without the change. However, for such a simple method the regression should be detected for sure.

**Comment by danmosemsft** [comment]
>  the perf characteristics of this method are size-independent

If weuse whatever the most commonly used size is (don't know what ) then the test will tell us if that accidentally changes.

**Pull request merged**